test/unit/test_http_utils.py

```diff
@@ -138,3 +138,20 @@ def test_discard_buffering_arg_on_py26(
 
         mock_http_response_init.assert_called_once_with(
             response_obj, **kwargs)
+
+    @contextlib.contextmanager
+    def _make_socket(self):
```

Comment:
Instead of creating this context manager and using it, you could simply use `contextlib.closing` instead, no need for a special method and such.